addons/account/models/account_move_line.py

```diff
@@ -2344,9 +2344,33 @@ def remove_move_reconcile(self):
     # ANALYTIC
     # -------------------------------------------------------------------------
 
+    def _validate_distribution(self):
+        for line in self.filtered(lambda line: line.display_type == 'product'):
+            mandatory_plans_ids = [plan['id'] for plan in self.env['account.analytic.plan'].sudo().get_relevant_plans(**{
+                        'product': line.product_id.id,
+                        'account': line.account_id.id,
+                        'business_domain': line.move_id.move_type in ['out_invoice', 'out_refund', 'out_receipt'] and 'invoice'
+                                           or line.move_id.move_type in ['in_invoice', 'in_refund', 'in_receipt'] and 'bill'
+                                           or 'general'
+                        }) if plan['applicability'] == 'mandatory']
+            if not mandatory_plans_ids:
+                continue
+            distribution_by_root_plan = {}
+            for analytic_account_id, percentage in (line.analytic_distribution or {}).items():
+                root_plan = self.env['account.analytic.account'].browse(analytic_account_id).root_plan_id
+                distribution_by_root_plan[root_plan.id] = distribution_by_root_plan.get(root_plan.id, 0) + percentage
+
+            for plan_id in mandatory_plans_ids:
+                if float_compare(distribution_by_root_plan.get(plan_id, 0), 100, precision_digits=2) != 0:
+                    raise ValidationError(_("A line with %saccount: '%s' has an invalid analytic distribution.",
+                                    _("product: '%s' and ", line.product_id.display_name) if line.product_id else "",
+                                    line.account_id.display_name))
```

Comment:
This is bad for translations, don't translate parts of sentences
https://www.odoo.com/documentation/15.0/developer/howtos/translations.html?highlight=translation#blocks

Reply:
ok